server/zoom/client.go

```diff
@@ -6,118 +6,134 @@ package zoom
 
-	out, err := token.SignedString([]byte(c.ApiSecret))
+	out, err := token.SignedString([]byte(c.apiSecret))
 	if err != nil {
 		return "", err
 	}
 
 	return out, nil
 }
 
-func closeBody(r *http.Response) {
-	if r.Body != nil {
-		ioutil.ReadAll(r.Body)
-		r.Body.Close()
-	}
-}
-
-type ClientError struct {
-	StatusCode int
-	Err        string
-}
-
-func (ce *ClientError) Error() string {
-	return ce.Err
-}
-
 func (c *Client) request(method string, path string, data interface{}, ret interface{}) *ClientError {
 	jsonData, err := json.Marshal(data)
 	if err != nil {
-		return &ClientError{http.StatusInternalServerError, err.Error()}
+		return &ClientError{http.StatusInternalServerError, err}
 	}
 
-	rq, err := http.NewRequest(method, c.BaseUrl+path, bytes.NewReader(jsonData))
+	rq, err := http.NewRequest(method, c.baseURL+path, bytes.NewReader(jsonData))
 	if err != nil {
-		return &ClientError{http.StatusInternalServerError, err.Error()}
+		return &ClientError{http.StatusInternalServerError, err}
 	}
 	rq.Header.Set("Content-Type", "application/json")
 	rq.Close = true
 
 	token, err := c.generateJWT()
 	if err != nil {
-		return &ClientError{http.StatusInternalServerError, err.Error()}
+		return &ClientError{http.StatusInternalServerError, err}
 	}
 	rq.Header.Set("Authorization", "BEARER "+token)
 
-	if rp, err := c.HttpClient.Do(rq); err != nil {
-		return &ClientError{http.StatusInternalServerError, fmt.Sprintf("Unable to make request to %v: %v", c.BaseUrl+path, err.Error())}
-	} else if rp == nil {
-		return &ClientError{http.StatusInternalServerError, fmt.Sprintf("Received nil response when making request to %v", c.BaseUrl+path)}
-	} else if rp.StatusCode >= 300 {
-		defer closeBody(rp)
-		buf := new(bytes.Buffer)
-		buf.ReadFrom(rp.Body)
-		return &ClientError{rp.StatusCode, buf.String()}
-	} else {
-		defer closeBody(rp)
-		buf := new(bytes.Buffer)
-		buf.ReadFrom(rp.Body)
-		if err := json.Unmarshal(buf.Bytes(), &ret); err != nil {
-			return &ClientError{rp.StatusCode, err.Error()}
+	rp, err := c.httpClient.Do(rq)
+	if err != nil {
+		return &ClientError{
+			http.StatusInternalServerError,
+			errors.WithMessagef(err, "Unable to make request to %v", c.baseURL+path),
+		}
+	}
+
+	if rp == nil {
+		return &ClientError{
+			http.StatusInternalServerError,
+			errors.Errorf("Received nil response when making request to %v", c.baseURL+path),
 		}
-		return nil
+	}
+
+	defer closeBody(rp)
+	buf := new(bytes.Buffer)
+	if _, err = buf.ReadFrom(rp.Body); err != nil {
+		return &ClientError{
+			http.StatusInternalServerError,
+			errors.Errorf("Failed to read response from %v", c.baseURL+path),
+		}
+	}
+
+	if rp.StatusCode >= 300 {
+		return &ClientError{rp.StatusCode, errors.New(buf.String())}
+	}
+
+	if err := json.Unmarshal(buf.Bytes(), &ret); err != nil {
+		return &ClientError{rp.StatusCode, err}
+	}
+
+	return nil
+}
+
+func closeBody(r *http.Response) {
+	if r.Body != nil {
+		ioutil.ReadAll(r.Body)
+		r.Body.Close()
```

Comment:
I don't know if this method is needed.

Reply:
Yes, responses for requests open by a client need to be closed, explicitly, even for a Keep-alive (the http stack manages the connections itself)